nobrainer/processing/segmentation.py

```diff
@@ -74,8 +82,13 @@ def _compile():
             )
 
         if warm_start:
+            if checkpoint_tracker:
+                self = checkpoint_tracker.load()
```

Comment:
shouldn't this be a static method or classmethod if it overwrites `self`? perhaps take a look at the save/load method in base estimator.